processor/resourcedetectionprocessor/internal/aws/ec2/ec2.go

```diff
@@ -88,13 +88,13 @@ func (d *Detector) Detect(ctx context.Context) (resource pcommon.Resource, schem
 		client := getHTTPClientSettings(ctx, d.logger)
 		tags, err := connectAndFetchEc2Tags(meta.Region, meta.InstanceID, d.tagKeyRegexes, client)
 		if err != nil {
-			return res, "", fmt.Errorf("failed fetching ec2 instance tags: %w", err)
```

Comment:
Isn't this already emitting the detected resource attributes in addition to the error?  Is the problem that it also needs to emit `conventions.SchemaURL` as the second return parameter, or maybe that the handling of this error at the caller should be adjusted?  The [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/sdk.md#detecting-resource-information-from-the-environment) is pretty clear that this should be treated as an error:

> Note the failure to detect any resource information MUST NOT be considered an error, whereas an error that occurs during an attempt to detect resource information SHOULD be considered an error.

In any event, tests are required for this change to assert the correct thing is happening and to guard against regression with future changes.

Reply:
Hey @Aneurysm9, the problem is the way the core resourcedetection component handles the returned resource when an error is returned alongside it, 
`	for _, detector := range p.detectors {
		r, schemaURL, err := detector.Detect(ctx)
		if err != nil {
			p.logger.Warn("failed to detect resource", zap.Error(err))
		} else {
			mergedSchemaURL = MergeSchemaURL(mergedSchemaURL, schemaURL)
			MergeResource(res, r, false)
		}
	}`

If it needs to be an error according to the spec, then I believe we need to change the way returned errors are handled to not ignore the attributes that were returned alongside the error, or alternatively maybe register the describeTags portion as it own detector

Reply:
And sure thing, I'll add test cases for any implementation we might go with

Reply:
I'm okay with moving forward with this change as-is (including tests).  We can re-evaluate the error handling behavior at a later date as I'm not sure there'd really be a functional difference.